fastlane/swift/SocketClient.swift

```diff
@@ -27,121 +27,121 @@ public enum SocketClientResponse: Error {
-        self.writeQueue = DispatchQueue(label: "writeQueue", qos: .background, attributes: .concurrent)
-        self.streamQueue = DispatchQueue.global(qos: .background)
-        self.socketStatus = .closed
+        readQueue = DispatchQueue(label: "readQueue", qos: .background, attributes: .concurrent)
+        writeQueue = DispatchQueue(label: "writeQueue", qos: .background, attributes: .concurrent)
+        streamQueue = DispatchQueue.global(qos: .background)
+        socketStatus = .closed
         self.socketDelegate = socketDelegate
         super.init()
     }
-    
+
     func connectAndOpenStreams() {
         var readStream: Unmanaged<CFReadStream>?
         var writeStream: Unmanaged<CFWriteStream>?
-        
-        self.streamQueue.sync {
+
+        streamQueue.sync {
             CFStreamCreatePairWithSocketToHost(kCFAllocatorDefault, self.host as CFString, self.port, &readStream, &writeStream)
-            
+
             self.inputStream = readStream!.takeRetainedValue()
             self.outputStream = writeStream!.takeRetainedValue()
-            
+
             self.inputStream.delegate = self
             self.outputStream.delegate = self
-            
+
             self.inputStream.schedule(in: .main, forMode: .defaultRunLoopMode)
             self.outputStream.schedule(in: .main, forMode: .defaultRunLoopMode)
         }
-        
-        self.dispatchGroup.enter()
-        self.readQueue.sync {
+
+        dispatchGroup.enter()
+        readQueue.sync {
             self.inputStream.open()
         }
-        
-        self.dispatchGroup.enter()
-        self.writeQueue.sync {
+
+        dispatchGroup.enter()
+        writeQueue.sync {
             self.outputStream.open()
         }
-        
+
         let secondsToWait = DispatchTimeInterval.seconds(SocketClient.connectTimeoutSeconds)
         let connectTimeout = DispatchTime.now() + secondsToWait
-        
-        let timeoutResult = self.dispatchGroup.wait(timeout: connectTimeout)
+
+        let timeoutResult = dispatchGroup.wait(timeout: connectTimeout)
         let failureMessage = "Couldn't connect to ruby process within: \(SocketClient.connectTimeoutSeconds) seconds"
-        
+
         let success = testDispatchTimeoutResult(timeoutResult, failureMessage: failureMessage, timeToWait: secondsToWait)
-        
+
         guard success else {
-            self.socketDelegate?.commandExecuted(serverResponse: .connectionFailure) { _ in }
+            socketDelegate?.commandExecuted(serverResponse: .connectionFailure) { _ in }
             return
         }
-        
-        self.socketStatus = .ready
-        self.socketDelegate?.connectionsOpened()
+
+        socketStatus = .ready
+        socketDelegate?.connectionsOpened()
     }
-    
+
     public func send(rubyCommand: RubyCommandable) {
         verbose(message: "sending: \(rubyCommand.json)")
         send(string: rubyCommand.json)
+        writeSemaphore.signal()
```

Comment:
This is the line that was causing the hang

Reply:
@janpio You're right I should've splitted the work in two tasks, but the fix was only one line, and the linting stuff was something it was in my head for a while now. I hope this serves. The explanation is more or less simple, there was a point were when the Ruby socket sent a failure and the Swift part tried both to write concurrently, causing a deadlock. This frees the write channel when that stack of functions is called.

Reply:
Wouldn't it be easier to quickly create a second PR for that 1 line change then? This PR will then be just about the linting after the other one is merged and this one rebased.